Replace sprintf usage with snprintf

zach2good opened the pull request:
Extracts one of the smaller items from this previous PR: https://github.com/mariadb-corporation/mariadb-connector-cpp/pull/16

License things (I saw some mention of this in other PRs):
The files I've modified are listed under LGPL v2.1, but I'm happy to hand over all rights to the maintainers and MariaDB group, to use in whatever way they like, under whatever license they want.

lawrinn commented on .gitignore:
```diff
@@ -39,6 +39,8 @@ benchmark/mysql-connector-cpp/
 benchmark/mariadb.json
 benchmark/mysql.json
 benchmark/main-benchmark
+cmake-build-*/
```
I gave some comments in the PR you referred. I'm gonna repeat them here.
These are too project-specific. I mean, I don't think they are default names of directories created by cmake/build by cmake generated projects.  I am not sure I want these to be added to .gitignore for everybody. On other hand it's pretty common naming pattern. Once again - I am not sure about this

lawrinn commented on src/logger/LoggerFactory.cpp:
```diff
@@ -25,7 +25,7 @@ namespace sql
 namespace mariadb
 {
-  Shared::Logger LoggerFactory::NO_LOGGER= (NO_LOGGER ? NO_LOGGER : Shared::Logger(new NoLogger()));
+  Shared::Logger LoggerFactory::NO_LOGGER= nullptr;
```
there is probably no need to assign it to nullptr - I think default constructor will do the same thing, but for cheaper.

lawrinn replied:
I've just pushed this line the way I want it(along with one change I needed). I suspect it should do the job with the warning

lawrinn commented on src/logger/LoggerFactory.cpp:
```diff
@@ -51,7 +51,7 @@ namespace mariadb
   {
     if (!NO_LOGGER) {
-      NO_LOGGER.reset(new NoLogger());
+      NO_LOGGER = Shared::Logger(new NoLogger());
```
In that PR I asked "Is there real need for this change? I don't think so" Now I guess the idea is to initialize in case it's not initialized. Like calling method of not initialized object is not a good idea. But it's not initialization and one line before we test that object as boolean and that involves method calling.
In 1.1 this dance around NO_LOGGER has been removed and I should do the same in 1.0 probably. But as for your  PR  - I guess there is probably no need to change this line

lawrinn commented on src/logger/LoggerFactory.h:
```diff
@@ -32,9 +32,11 @@ namespace mariadb
 {
 class LoggerFactory
 {
+private:
```
why is that? also not needed. Everything is private in a class by default

lawrinn commented:
Thank you for another pull request. I've made my comments.
Also please state again, that you are submitting your code under the BSD-new license. I am afraid that needs to be done for each PR

zach2good commented:
Thanks for following up and pushing that logging change into `master`! I've stripped down this PR so now it's just that `snprintf` change, and I've added the license stuff to the main post 👍

lawrinn commented:
Out of curiosity - what do you use it for? also, since your patch against master that is 1.0 version, are you using 1.0 and if yes why not the 1.1? It is also ga, currently it is still in develop branch. To my taste it's a way better

zach2good commented:
Ah, I assumed that `master` was whatever the running current release cycle
was. I'll probably move over to using `develop` soon then!

I do a lot of the engine/core work for `Final Fantasy XI` MMO server
emulators. I introduced the cpp connector over our existing use of the c
connector to make it easier
to support prepared statements, make result access thread-safe, and be able
to pull out results by column name from result sets. The change was
initially born out of
trying to squeeze out some more performance through prepared statements,
but all the extra goodies have been great!

This is the PR I added these changes in:
https://github.com/LandSandBoat/server/pull/4601

This is the layer I wrap the cpp connector with:
https://github.com/LandSandBoat/server/blob/base/src/common/database.h#L168

I think my only issues have been having to reimplement or workaround some
things that are missing in the cpp connector that I was using in the c
connector, like `affectedRows`:
https://github.com/LandSandBoat/server/blob/base/src/common/database.h#L225

But largely the experience has been positive, thanks for you work :+1:

On Tue, Nov 12, 2024 at 5:04 PM lawrinn ***@***.***> wrote:

> Out of curiosity - what do you use it for? also, since your patch against
> master that is 1.0 version, are you using 1.0 and if yes why not the 1.1?
> It is also ga, currently it is still in develop branch. To my taste it's a
> way better
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mariadb-corporation/mariadb-connector-cpp/pull/21#issuecomment-2471099938>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKTJIIZUMKDGLWPFJ4MNP32AIYI5AVCNFSM6AAAAABRAEYKEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZRGA4TSOJTHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

zach2good commented:
Oh, I did a writeup about this upgrade process you might find interesting:
https://github.com/LandSandBoat/server/wiki/Database-Library-Upgrade

On Tue, Nov 12, 2024 at 5:15 PM Zach Toogood ***@***.***> wrote:

> Ah, I assumed that `master` was whatever the running current release cycle
> was. I'll probably move over to using `develop` soon then!
>
> I do a lot of the engine/core work for `Final Fantasy XI` MMO server
> emulators. I introduced the cpp connector over our existing use of the c
> connector to make it easier
> to support prepared statements, make result access thread-safe, and be
> able to pull out results by column name from result sets. The change was
> initially born out of
> trying to squeeze out some more performance through prepared statements,
> but all the extra goodies have been great!
>
> This is the PR I added these changes in:
> https://github.com/LandSandBoat/server/pull/4601
>
> This is the layer I wrap the cpp connector with:
> https://github.com/LandSandBoat/server/blob/base/src/common/database.h#L168
>
> I think my only issues have been having to reimplement or workaround some
> things that are missing in the cpp connector that I was using in the c
> connector, like `affectedRows`:
> https://github.com/LandSandBoat/server/blob/base/src/common/database.h#L225
>
> But largely the experience has been positive, thanks for you work :+1:
>
> On Tue, Nov 12, 2024 at 5:04 PM lawrinn ***@***.***> wrote:
>
>> Out of curiosity - what do you use it for? also, since your patch against
>> master that is 1.0 version, are you using 1.0 and if yes why not the 1.1?
>> It is also ga, currently it is still in develop branch. To my taste it's a
>> way better
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mariadb-corporation/mariadb-connector-cpp/pull/21#issuecomment-2471099938>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAKTJIIZUMKDGLWPFJ4MNP32AIYI5AVCNFSM6AAAAABRAEYKEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZRGA4TSOJTHA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> com>
>>
>

lawrinn commented:
Well, 1.1. was supposed to become master once it goes ga, but then I've become reluctant of doing this pull requests are one of reasons - people submit their requests against master, and it's a way easier to merge from 1.0 to 1.1. then in opposite direction. Asking to rebase against different branch makes PR's a bit more cumbersome.
I didn't quite get your affected rows problem. Looking at the code it seems like you you are getting total number of rows in the resultset. But resultset has rowsCount() method. If you don't use resultset streaming it will give you the number of rows in the resultset object, otherwise it will be number of rows fetched at a time. I guess this is because of lack of documentation. In most cases you can refer to JDBC specification. The API is mostly following it
Thank you, it was interesting to read!

lawrinn commented:
hmm, it looks like you use rowsCount(). if (rset && rset->rowsCount() && rset->next())
also, that is redundant to check here rowsCount() imho.
